zerver/lib/export.py

```diff
@@ -1380,7 +1385,15 @@ def import_uploads_local(import_dir: Path, processing_avatars: bool=False) -> No
             else:
                 file_path += '.png'
         else:
-            file_path = os.path.join(settings.LOCAL_UPLOADS_DIR, "files", record['s3_path'])
+            # Should be kept in sync with its equivalent in zerver/lib/uploads in the
+            # function 'upload_message_image'
+            s3_file_name = "/".join([
+                str(record['realm_id']),
+                random_name(18),
+                sanitize_name(os.path.basename(record['path']))
+            ])
+            file_path = os.path.join(settings.LOCAL_UPLOADS_DIR, "files", s3_file_name)
```

Comment:
In importing avatars, we use the implementation where the 'avatar_path'
is seperately calculated using realm and user ID and then the content
of the path provided in the avatar's 'records.json' are copied to this
'avatar_path'.

Similary, here for the uploads, 's3_file_name' is seperately calculated
using the realm ID and uploaded file name and then the content of the
path provided in upload's 'records.json' are copied to this 's3_file_name'.